pkg/operator/ceph/cluster/cluster.go

```diff
@@ -146,19 +153,46 @@ func (c *cluster) doOrchestration(rookImage string, cephVersion cephver.CephVers
 	}
 
 	// Start Ceph manager
-	mgrs := mgr.New(c.Info, c.context, c.Namespace, rookImage,
-		spec.CephVersion, cephv1.GetMgrPlacement(spec.Placement), cephv1.GetMgrAnnotations(c.Spec.Annotations),
-		spec.Network, spec.Dashboard, spec.Monitoring, spec.Mgr, cephv1.GetMgrResources(spec.Resources),
-		cephv1.GetMgrPriorityClassName(spec.PriorityClassNames), c.ownerRef, c.Spec.DataDirHostPath, c.Spec.SkipUpgradeChecks)
+	mgrs := mgr.New(c.Info,
+		c.context,
+		c.Namespace,
+		rookImage,
+		spec.CephVersion,
```

Comment:
We should really pass the whole `spec` instead of passing so many different fields of the spec. This is getting ridiculous with all these args. For another day...

Reply:
Agreed, the mon code does pass the whole spec, and that's much better.